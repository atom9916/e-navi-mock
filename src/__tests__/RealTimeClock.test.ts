/**
 * @jest-environment jsdom
 */

import { render, screen } from '@testing-library/vue'
import RealTimeClockVue from '@/components/RealTimeClock.vue'
import { mount } from '@vue/test-utils'

describe('ReallTimeClockコンポーネントのテスト', () => {
  beforeEach(() => {
    jest.useFakeTimers()
  })
  afterEach(() => {
    jest.useRealTimers()
  })
  test('現在の日付と時刻が正しく表示される', async () => {
    jest.setSystemTime(new Date(2023, 5, 26, 9, 0, 0))

    const now = new Date()
    const expectedCurrentDate = now.toLocaleDateString()
    const expectedCurrentTime = now.toLocaleTimeString()

    const wrapper = mount(RealTimeClockVue)
    await wrapper.vm.$nextTick()

    const currentDate = wrapper.find('p#today-contents').text()
    const currentTime = wrapper.find('p#now-contents').text()

    expect(currentDate).toEqual(`今日は\u00A0${expectedCurrentDate}\u00A0です`)
    expect(currentTime).toEqual(`現在の時刻は\u00A0${expectedCurrentTime}\u00a0です`)
  })

  test('スナップショットテスト', () => {
    const { container } = render(RealTimeClockVue)

    expect(container).toMatchSnapshot()
  })
})
