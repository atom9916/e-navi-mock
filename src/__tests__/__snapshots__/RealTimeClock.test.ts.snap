// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReallTimeClockコンポーネントのテスト スナップショットテスト 1`] = `
<div>
  
  <section
    aria-labelledby="today-contents"
  >
    <p
      id="today-contents"
    >
       今日は 
      <span
        style="font-weight: 500; font-size: 1.4em; color: red;"
      />
       です 
    </p>
  </section>
  <section
    aria-labelledby="now-contents"
  >
    <p
      id="now-contents"
    >
       現在の時刻は 
      <span
        style="font-weight: 250; font-size: 1.4em;"
      />
       です 
    </p>
  </section>
  <section
    aria-labelledby="encouragement-message"
  >
    <p
      id="encouragement-message"
    >
      今日もお仕事、がんばりましょう…
    </p>
  </section>
  
</div>
`;
